ext/opentelemetry-ext-mysql/src/opentelemetry/ext/mysql/__init__.py

```diff
@@ -41,23 +41,32 @@
 ---
 """
 
+import typing
+
 import mysql.connector
 
-from opentelemetry.ext.dbapi import trace_integration as db_integration
-from opentelemetry.trace import Tracer
+from opentelemetry.ext.dbapi import wrap_connect
+from opentelemetry.ext.mysql.version import __version__
+from opentelemetry.trace import TracerProvider, get_tracer_provider
 
 
-def trace_integration(tracer: Tracer):
+def trace_integration(tracer_provider: typing.Optional[TracerProvider] = None):
     """Integrate with MySQL Connector/Python library.
        https://dev.mysql.com/doc/connector-python/en/
     """
+
+    if tracer_provider is None:
```

Comment:
any reason not to call the original `db_integration` with the tracer_provider as a parameter since it's doing the same check?

Reply:
The problem is that it'll create the tracer with the version and name from the db-api module. I had to split the logic from `opentelemetry.ext.dbapi.trace_integration` into two different functions, one receiving a tracer provider (to be used by the user), and other one receiving a tracer (to be used by other integrations).